gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -0,0 +1,292 @@
+   *
+   * <p>Google Cloud DNS currently supports records of type A, AAAA, CNAME, MX NAPTR, NS, PTR, SOA,
+   * SPF, SRV, TXT.
+   *
+   * @see <a href="https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types">Cloud DNS
+   * supported record types</a>
+   */
+  public enum DnsRecordType {
+    /**
+     * Address record, which is used to map host names to their IPv4 address.
+     */
+    A,
+    /**
+     * IPv6 Address record, which is used to map host names to their IPv6 address.
+     */
+    AAAA,
+    /**
+     * Canonical name record, which is used to alias names.
+     */
+    CNAME,
+    /**
+     * Mail exchange record, which is used in routing requests to mail servers.
+     */
+    MX,
+    /**
+     * Naming authority pointer record, defined by RFC3403.
+     */
+    NAPTR,
+    /**
+     * Name server record, which delegates a DNS zone to an authoritative server.
+     */
+    NS,
+    /**
+     * Pointer record, which is often used for reverse DNS lookups.
+     */
+    PTR,
+    /**
+     * Start of authority record, which specifies authoritative information about a DNS zone.
+     */
+    SOA,
+    /**
+     * Sender policy framework record, which is used in email validation systems.
+     */
+    SPF,
+    /**
+     * Service locator record, which is used by some voice over IP, instant messaging protocols and
+     * other applications.
+     */
+    SRV,
+    /**
+     * Text record, which can contain arbitrary text and can also be used to define machine readable
+     * data such as security or abuse prevention information.
+     */
+    TXT
+  }
+
+  public static class Builder {
+
+    private List<String> rrdatas = new LinkedList<>();
+    private String name;
+    private Integer ttl;
+    private DnsRecordType type;
+
+    private Builder() {
+    }
+
+    /**
+     * Creates a builder and pre-populates attributes with the values from the provided DnsRecord
+     * instance.
+     */
+    private Builder(DnsRecord record) {
+      this.name = record.name;
+      this.ttl = record.ttl;
+      this.type = record.type;
+      this.rrdatas.addAll(record.rrdatas);
+    }
+
+    /**
+     * Adds a record to the record set. The records should be as defined in RFC 1035 (section 5) and
+     * RFC 1034 (section 3.6.1). Examples of records are available in Google DNS documentation.
+     *
+     * @see <a href="https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types">Google
+     * DNS documentation </a>.
+     */
+    public Builder addRecord(String record) {
+      this.rrdatas.add(checkNotNull(record));
+      return this;
+    }
+
+    /**
+     * Removes a record from the set. An exact match is required.
+     */
+    public Builder removerRecord(String record) {
+      this.rrdatas.remove(checkNotNull(record));
+      return this;
+    }
+
+    /**
+     * Removes a record on the given index from the set.
+     */
+    public Builder removerRecord(int index) {
```

Comment:
It might be useful to have the remove based on the record rather than the index.